stdlib/test/builtin/test_string.mojo

```diff
@@ -594,11 +594,58 @@ fn test_rfind() raises:
 
 
 fn test_split() raises:
-    # Reject empty delimiters
-    with assert_raises(
-        contains="empty delimiter not allowed to be passed to split."
-    ):
-        _ = String("hello").split("")
+    # test all utf8 and unicode separators
+    var _line_sep_utf8 = List[UInt8](0x20, 0x5C, 0x75, 0x32, 0x30, 0x32, 0x38)
+    """Unicode Line Separator: \\u2028."""
+    var _paragraph_sep_utf8 = List[UInt8](
+        0x20, 0x5C, 0x75, 0x32, 0x30, 0x32, 0x39
+    )
+    """Unicode Paragraph Separator: \\u2029."""
+    # TODO add line and paragraph separator as stringliteral once unicode escape secuences are accepted
+    var _universal_separators = List[String](
+        String("\n"),
+        String("\r"),
+        String("\v"),
+        String("\f"),
+        String("\x1c"),
+        String("\x1e"),
+        String("\x85"),
+        String(_line_sep_utf8),
+        String(_paragraph_sep_utf8),
+    )
+    for i in _universal_separators:
+        assert_true(i[].isspace())
+
+    for i in List[String]("not", "space", "", "s", "a", "c"):
+        assert_false(i[].isspace())
+
+    # empty separators default to whitespace
+    var d = String("hello world").split("")
+    assert_true(d[0] == "hello", d[1] == "world")
+    d = String("hello \t\n\n\v\fworld").split("\n")
+    assert_true(d[0] == "hello \t" and d[1] == "" and d[2] == "\v\fworld")
```

Comment:
While explicitly tests are good, a great way to improve coverage for this would be to check for the invariant `sep.join(some_str.split(sep)) == some_str`.